Product/Production/Services/DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/DSDeferredResponseAuditTransformsTest.java

```diff
@@ -0,0 +1,216 @@
+    public void transformRequestToAuditMsg() throws ConnectionManagerException, UnknownHostException, JAXBException {
+
+        DSDeferredResponseAuditTransforms transforms = new DSDeferredResponseAuditTransforms() {
+
+            @Override
+            protected String getLocalHostAddress() {
+                return REQUEST_LOCAL_IP;
+            }
+
+            @Override
+            protected String getWebServiceUrlFromRemoteObject(NhinTargetSystemType target, String serviceName) {
+                return REQUEST_WS_REQUEST_URL;
+            }
+
+            @Override
+            protected String getDSDeferredRequestInitiatorAddress(NhinTargetSystemType target) {
+                return REQUEST_REMOTE_IP;
+            }
+        };
+
+        RegistryResponseType request = new RegistryResponseType();
+
+        AssertionType assertion = createAssertion();
+        LogEventRequestType auditRequest = transforms.transformRequestToAuditMsg(request, assertion,
+            createNhinTarget(), NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION, NhincConstants.AUDIT_LOG_NHIN_INTERFACE,
+            Boolean.TRUE, null, NhincConstants.NHINC_XDR_RESPONSE_SERVICE_NAME);
+        testGetEventIdentificationType(auditRequest, NhincConstants.NHINC_XDR_RESPONSE_SERVICE_NAME, Boolean.TRUE);
+        testCreateActiveParticipantFromUser(auditRequest, Boolean.TRUE, assertion);
+        testGetActiveParticipantDestination(auditRequest, Boolean.TRUE, null, REQUEST_WS_REQUEST_URL, REQUEST_LOCAL_IP);
+        testAuditSourceIdentification(auditRequest.getAuditMessage().getAuditSourceIdentification(), assertion);
+        testGetActiveParticipantSource(auditRequest, Boolean.TRUE, null, REQUEST_REMOTE_IP);
+        assertEquals("AuditMessage.Request direction mismatch", auditRequest.getDirection(),
+            NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION);
+        assertEquals("AuditMessage.Request interface mismatch", auditRequest.getInterface(),
+            NhincConstants.AUDIT_LOG_NHIN_INTERFACE);
+    }
+
+    @Override
+    protected void testCreateActiveParticipantFromUser(LogEventRequestType request, Boolean isRequesting,
+        AssertionType assertion) {
+
+        if (isRequesting) {
+            AuditMessageType.ActiveParticipant userActiveParticipant = null;
+            List<AuditMessageType.ActiveParticipant> activeParticipant = request.getAuditMessage().getActiveParticipant();
+            for (AuditMessageType.ActiveParticipant item : activeParticipant) {
+                // TODO: Where does this hard-coded value of "Code" come from?
+                if (item.getRoleIDCode().get(0).getCode() != null && item.getRoleIDCode().get(0).getCode()
+                    .equals("Code")) {
+
+                    userActiveParticipant = item;
+                    break;
+                }
+            }
+
+            assertNull("userActiveParticipant is null", userActiveParticipant);
+        }
+
+        // TODO: Should there be an else case here?
+    }
```

Comment:
We can remove the "TODO: Should there be an else case here?" as there is no else case for DSdeferredResponse